guides/how-to-decaf.md

```diff
@@ -0,0 +1,79 @@
+
+With those rules/recommendations out of the way, following below is the general guide that should be followed:
+
+1. As you might expect, decide what you are going to decaffeinate. To help make everyone's lives easier, you should ensure the file you are decaffing doesn't have any open PRs against it, or be at risk of receiving major changes anytime soon. If you aren't sure of this, feel free to ask the Pulsar maintainers.
+
+2. Once you've decided what repo/package you are going to decaf, then decide if you want to decaf the specs or source files.
+
+3. Now with the choices out of the way, you can get started. You'll want to make sure to create the Machine Decaf branch (more on that later) but this would be a branch that exists on the remote repo that contains only machine decaffed files. (If you don't have permissions to make this, feel free to ask the Pulsar maintainers). While we could be done there, the machine decaffed code is valid, but very unreadable and difficult to work with, so it requires manual intervention to improve.
+
+4. To start you manual decaffeination, create a fork of the machine decaf files branch, and apply any changes needed on top of it (more on that later). Once you've applied all changes you need to, to improve the machine decaf work you can finally submit a PR.
+
+5. When submitting the PR of your manual decaf work, ensure the target branch is the machine decaf branch. Having this be the target allows maintainers to quickly and painlessly review your work, and check for any actual changes that you've made.
+
+6. Finally, once your manual decaf work is approved, it should be merged directly into the `master`/`main` branch on that repo. So that your decaf changes are applied directly to the source. From here you can cleanup by deleting the machine decaf branch.
+
+## Machine Decaf
+
+Like mentioned earlier, using automated tools to decaf CoffeeScript produces valid, yet unreadable and unmaintainable code.
+
+To produce this new code in a way that we know is functional, you'll want to keep lookout for a few things:
+
+First note, that in general using CoffeeScript 2.x (the latest version as of this writing) is totally valid. But if you find any tests are failing with only machine decaf code, it may be because there are very few edge cases where CoffeeScript code decafs differently from 2.x tooling and 1.x tooling. Which because nearly all the CoffeeScript for Atom was written back in the CoffeeScript 1.x days, we need to account for this. You can always check if a particular repo/package is using 1.x CoffeeScript tooling by checking the version of `coffee-script` or `coffeescript` within the relevant `package.json` either `dependencies` or `devDependencies`. If those are within the 1.x range, you should use older tooling to decaf. While keeping in mind that the 1.x CoffeeScript decaf tooling will usually strip any comments from the code, and will have to be added back in.
```

Comment:
```suggestion
First note, that in general using CoffeeScript 2.x (the latest version as of this writing) is totally valid. But if you find any tests are failing with only machine decaf code, it may be because there are very few edge cases where CoffeeScript code decafs differently from 2.x tooling and 1.x tooling. Which, because nearly all the CoffeeScript for Atom was written back in the CoffeeScript 1.x days, we need to account for. You can always check if a particular repo/package is using 1.x CoffeeScript tooling by checking the version of `coffee-script` or `coffeescript` within the relevant `package.json` either `dependencies` or `devDependencies`. If those are within the 1.x range, you should use older tooling to decaf. While keeping in mind that the 1.x CoffeeScript decaf tooling will usually strip any comments from the code, and will have to be added back in.
```